plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -0,0 +1,159 @@
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.json.JsonXContent;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.sdk.DeleteDataObjectRequest;
+import org.opensearch.sdk.DeleteDataObjectResponse;
+import org.opensearch.sdk.GetDataObjectRequest;
+import org.opensearch.sdk.GetDataObjectResponse;
+import org.opensearch.sdk.PutDataObjectRequest;
+import org.opensearch.sdk.PutDataObjectResponse;
+import org.opensearch.sdk.SdkClient;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.DeleteItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+
+/**
+ * DDB implementation of {@link SdkClient}. DDB table name will be mapped to index name.
+ *
+ */
+@AllArgsConstructor
+@Log4j2
+public class DDBOpenSearchClient implements SdkClient {
+
+    private static final String DEFAULT_TENANT = "DEFAULT_TENANT";
+
+    private static final String HASH_KEY = "tenant_id";
+    private static final String RANGE_KEY = "id";
+    private static final String SOURCE = "source";
+
+    private DynamoDbClient dynamoDbClient;
+
+    /**
+     * DDB implementation to write data objects to DDB table. Tenant ID will be used as hash key and document ID will
+     * be used as range key. If tenant ID is not defined a default tenant ID will be used. If document ID is not defined
+     * a random UUID will be generated. Data object will be written as a nested DDB attribute.
+     *
+     */
+    @Override
+    public CompletionStage<PutDataObjectResponse> putDataObjectAsync(PutDataObjectRequest request, Executor executor) {
+        final String id = request.id() != null ? request.id() : UUID.randomUUID().toString();
+        final String tenantId = request.tenantId() != null ? request.tenantId() : DEFAULT_TENANT;
```

Comment:
curious if `DEFAULT_TENANT` should be in the request object to start with.

Reply:
`DEFAULT_TENANT` will not be required for other data store implementations. Given that it is going to be specific to DDB store I would prefer it to be part of `DDBOpenSearchClient.java`